File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    
    horizon = []
    vertical = []
    
    for size in sizes:
        if size[0] >= size[1]:
            horizon.append(size[0])
            vertical.append(size[1])
        else:
            horizon.append(size[1])
            vertical.append(size[0])
    answer = max(horizon) * max(vertical)
    return answer

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        
        var widthList = new ArrayList<Integer>();
        var heightList = new ArrayList<Integer>();
        
        for (var s: sizes) {
            if (s[0] > s[1]) {
                widthList.add(s[0]);
                heightList.add(s[1]);
            } else {
                widthList.add(s[1]);
                heightList.add(s[0]);
            }
        }
        
        return Collections.max(widthList) * Collections.max(heightList);
    }
}